### DISCUSSION TITLE
Q: The future of React Aria vs Spectrum Web Components?

### DISCUSSION THREAD

**Discussion by kkimdev** [adobe/react-spectrum#7445] [Q&A] [answered]
Hi,

My understanding is that [Spectrum Web Components](https://github.com/adobe/spectrum-web-components) will be the next generation design library for Adobe.  Does that mean React Spectrum and React Aria could be deprecated in the coming years?  Or will there be a next version of React Aria, like Aria Web Components?

**Comment by dannify** [comment, marked as answer]
Very incorrect! The Spectrum Web Components library is just another library supporting the Spectrum design system. We work with them to align our components. React Spectrum and React Aria will continue to be built and improved upon. Currently we are working on the new Spectrum design update, Spectrum 2, with no signs of stopping!

**Reply by mvelbaum** [reply]
@dannify can you shed some light on the reason why you guys maintain and develop both a framework agnostic web components implementation and a react-specific one?

1. If I were using React, is there a reason to go with the WC library?
2. Is there feature/component parity between React and WC libraries?
3. Does Adobe themselves prefer to use WC in their new products?

Thanks!

**Reply by snowystinger** [reply]
Adobe is a large company and we have applications built in many different frameworks. The most common of which is React. As a result, we have multiple implementations of our design system.

1. If you're using React, I'd suggest using this implementation, as it was developed explicitly for React. Web components have historically had problems living in a React app and React 19 is the first time they have been included as first class citizens. It may be easier to mix and match going forward, however, there's still an advantage using a library written in the same framework. Many API decisions were made with React in mind and it's aware of the React lifecycle and context.

2. Because the libraries are built and maintained by different teams, there will be differences. That said, there is a lot of overlap as described by the design system itself and it should always be on a path of convergence.

3. It depends on what teams would already be familiar with and the ecosystem they live in.

**Comment by dannify** [comment]
@kkimdev Where did you get that idea?
